reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -231,11 +290,33 @@ private U mergedItem(@NonNull final U currentItem, @NonNull final U newItem) {
         return mergeValues(currentItem, newItem);
     }
 
-    protected void put(@NonNull final U item, @NonNull final Uri uri) {
+    @NonNull
+    protected Single<Boolean> put(@NonNull final Uri uri, @NonNull final U item) {
```

Comment:
Put and delete method are really similar, you can somehow extract duplications into one method